torch/ao/quantization/observer.py

```diff
@@ -1092,15 +1092,17 @@ def _adjust_min_max(
         # Therefore, we avoid such division operation.
         downsample_rate = int(
             torch.ceil(
-                (combined_max - combined_min) * upsample_rate / (self.max_val - self.min_val)
+                ((combined_max - combined_min) / (self.max_val - self.min_val)) * upsample_rate
             ).item()
         )
-        e = downsample_rate * (self.max_val - self.min_val) / upsample_rate - (combined_max - combined_min)
+        e = downsample_rate / upsample_rate * (self.max_val - self.min_val) - (combined_max - combined_min)
         start_idx = int(
-            torch.round((self.min_val - combined_min) * self.bins * upsample_rate / (self.max_val - self.min_val)).item()
+            torch.round((self.min_val - combined_min) / (self.max_val - self.min_val) * self.bins * upsample_rate).item()
         )
         combined_max = combined_max + e
         combined_min = combined_min
+        assert not torch.isinf(torch.tensor(combined_max - combined_min)), "combined min/max are inf"
+        assert not torch.isinf(torch.tensor(downsample_rate)), "downsample"
```

Comment:
nit: seems error message is not complete

Reply:
yeah the PR is not passing CI and I don't know why. The tests pass on my machine but not in CI. Any advice would be appreciated, its the new 'test close to infinity' test that's failing too.